Fix issue with rendering pre-rendered and instant pages.

yousuf811 opened the pull request:
Seems like we had a typo in the webRequest.onCompleted event handler.

This caused plenty of issue but most significantly affected prerendered
and instant pages in chrome.

You can reproduce this error in multiple ways:

1. Open up chrome and type in guardian.co.uk in the address bar but don't hit enter.
2. Wait 5 seconds. (this will ensure page is prerendered in the
background.)
3. Now hit enter.

Notice the clacks icon doesn't show up.

Since prerendered pages are hidden they have a tabId that ends up raising an error on
pageAction.show and hide. I've added safe versions of them.

This problem is significant because Google has started to add the gnu
header and nobody has noticed this because of the bug.

To check this out just search for terry pratchett on google with this
patch committed and you'll see the clacks popup.

newfolder0 commented:
Wow, that's a very awesome thing for Google to do. I am consistently amazed at the places where the clacks header pops up, I even received a bug report from someone who thought it must have been a mistake when they saw it while buying shoes online.

Thanks a lot for fixing the typo, that certainly explains a few things! Good work with the preloading issue too, I hadn't noticed it and likely wouldn't have any time soon. I really appreciate that you actually went ahead and fixed things instead of just submitting a bug report, I know that's what Github is for but I appreciate it nonetheless. I just haven't had any time to put into this over the last couple of weeks and this will probably resolve a few of the bug reports I hadn't managed to pin down! Perfect!

yousuf811 commented:
Thanks for merging the pull request. When do you think you'll be able to push a new release out.

newfolder0 commented:
As soon as I work out what's going on with Git... I merged the request and then tried to pull it locally and somehow, somewhere along the way it was lost again. src/background.js doesn't have your changes now. I don't know yet what I did wrong, sorry, I would have worked it out last night but someone showed up for dinner.

newfolder0 commented:
Dunno what happened there but once I reverted background.js to the commit after the pull request, all was well. I've updated the Web Store version and it should available any minute now! Thank you very much again for your input, it is also nice that it now responds faster in places where it used to work once the page had finished loading.

yousuf811 commented:
Great. Thanks for pushing it out. I am glad it worked out. And thanks for
crediting me on the README.

I am trying to think of ways to avoid asking for the extra permissions
people are complaining about. Do you need that primarily for reading the
META tags ? Or is it required even for reading the headers.

On Sun, Apr 12, 2015 at 3:21 AM, newfolder0 notifications@github.com
wrote:

> Dunno what happened there but once I reverted background.js to the commit
> after the pull request, all was well. I've updated the Web Store version
> and it should available any minute now! Thank you very much again for your
> input, it is also nice that it now responds faster in places where it used
> to work once the page had finished loading.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/newfolder0/chrome-clacks/pull/8#issuecomment-92029924
> .
